### DISCUSSION TITLE
Training Overview - PostgresML

### DISCUSSION THREAD

**Discussion by giscus** [postgresml/postgresml#351] [Announcements]
# Training Overview - PostgresML

Train and deploy models to make online predictions using only SQL, with an open source Postgres extension.

https://postgresml.org/user_guides/training/overview/

**Comment by Prem-AU** [comment]
I am getting following error while loading the data in the pstgresql database table, can explain how to load the data to pgml to start training??

pgml_development=# select pgml.load_dataset('crowd_count');
WARNING:  Starting new HTTPS connection (1): huggingface.co:443
WARNING:  https://huggingface.co:443 "GET /api/datasets/crowd_count HTTP/1.1" 401 41
ERROR:  FileNotFoundError: Couldn't find a dataset script at /var/lib/postgresql/14/main/crowd_count/crowd_count.py or any data file in the same directory. Couldn't find 'crowd_count' on the Hugging Face Hub either: FileNotFoundError: Dataset 'crowd_count' doesn't exist on the Hub. If the repo is private, make sure you are authenticated with `use_auth_token=True` after logging in with `huggingface-cli login`.
CONTEXT:  Traceback (most recent call last):
  PL/Python function "load_dataset", line 5, in <module>
    return load(source, subset, limit, **json.loads(kwargs))
  PL/Python function "load_dataset", line 34, in load
  PL/Python function "load_dataset", line 43, in load_dataset
  PL/Python function "load_dataset", line 1674, in load_dataset
  PL/Python function "load_dataset", line 1451, in load_dataset_builder
  PL/Python function "load_dataset", line 1172, in dataset_module_factory
PL/Python function "load_dataset"

**Reply by montanalow** [reply]
There is no public dataset named 'crowd_count'. PostgresML does provide a few of the toy datasets from scikit learn ('digits', 'diabetes', 'iris', 'linerrud', 'wine', 'breast_cancer'), and we also support many of the datasets from [hugging face](https://huggingface.co/datasets). If you are trying to load a private dataset, I'd load it into the Postgres database using [COPY](https://www.postgresql.org/docs/current/sql-copy.html), or whatever is easiest with your framework/working language. 

The free hosted Gym also includes the ability to upload CSV files via the main navigation bar.
